universidades_d/dags/ETL_Universities_D.py

```diff
@@ -0,0 +1,109 @@
+import os
+from datetime import datetime, timedelta
+
+from airflow import DAG
+from airflow.operators.python import PythonOperator
+from dotenv import load_dotenv
+from sqlalchemy import create_engine
+
+# [END import_module]
+
+# [START default_args]
+# These args will get pass on to each operator
+# You can override them on a per-task basis during operator initialization
+default_args = {
+    'owner': 'alkymer',
+    'depends_on_past': False,
+    'email': ['juan.i.elizondo@hotmail.com'],
+    'email_on_failure': False,
+    'email_on_retry': False,
+    'retries': 10,
+    'retry_delay': timedelta(minutes=2),
+    'scheduler_interval': timedelta(hours=1)
+}
+
+
+# credentials for establish a database connection
+def get_connection_credentials():
+    load_dotenv()
+    pg_user = os.getenv('_PG_USER')
+    pg_passwd = os.getenv('_PG_PASSWD')
+    pg_host = os.getenv('_PG_HOST')
+    pg_port = os.getenv('_PG_PORT')
+    pg_db = os.getenv('_PG_DB')
+    return f'postgresql://{pg_user}:{pg_passwd}@{pg_host}:{pg_port}/{pg_db}'
+
+
+# [END default_args]
+class ETL:
+
+    # getting engine for plug connection and test status
+    def create_connection(self):
+        url = get_connection_credentials()
+        engine = create_engine(url)
+        return engine
+
+    # START extract function for get data from Database
+    def extract(self):
+        connection = self.create_connection()
+        connection_status = connection.connect()
+        connection_check = bool(connection_status)
+        if connection_check is False or not connection:
+            raise ValueError('Connection to database fails')
```

Comment:
No es necesario analizar if connection_check **is False**, si connection_check es bool, podría ser directamente if coneection_chek.
Por otro lado, si no se logra la conexión, se levanta un ValueError, creo que no es el tipo correcto de error a utilizar.